### PR TITLE
Refactor ENTSOE errors

### PR DESCRIPTION
**Pull request by systemcatch**
There was some discussion on slack about how error handling in the ENTSOE parser could be simplified.  I've created a new function that finds the error message and a new class of error for clarity.  I've not been able to really test this so you'll want to check carefully that it still functions as expected.

### CODE REVIEW

File: parsers/ENTSOE.py [hunk 1]
```diff
@@ -96,14 +113,10 @@ def query_consumption(domain, session, now=None):
...
-        # Grab the error if possible
-        soup = BeautifulSoup(response.text, 'html.parser')
-        text = soup.find_all('text')
-        if len(text):
-            error_text = soup.find_all('text')[0].prettify()
-            if 'No matching data found' in error_text: return
-            raise Exception('Failed to get consumption. Reason: %s' % error_text)
-    
+        check_page = find_query_error(response)
+        if check_page:return
```
Comment (corradio) [inline parsers/ENTSOE.py]: Why not raise the error inside the `find_query_error` method (which we should rename to `check_response`)?
A bit like the `raise_for_status` in python requests

Comment (systemcatch) [reply parsers/ENTSOE.py]: Sure I was just following the original format but that's easy to implement.

Comment (corradio) [reply parsers/ENTSOE.py]: thanks!

**Comment by corradio** [comment]
@systemcatch I think you should revert commit https://github.com/tmrowco/electricitymap/pull/798/commits/51a7e059af5ce33047bd75775f52433252a7ab3b (the merge). Somehow it went wrong.

**Comment by systemcatch** [comment]
@corradio ugh I was worried about that.  I'll try to fix but it might end up being quicker to create a new branch and resubmit the PR.

**Pull request closed without merging**